Sources/Ollama/Client.swift

```diff
@@ -302,17 +305,21 @@ extension Client {
     ///   - model: The name of the model to use for the chat.
     ///   - messages: The messages of the chat, used to keep a chat memory.
     ///   - options: Additional model parameters as specified in the Modelfile documentation.
+    ///   - tools: Override the list of tools to include in this chat.
     ///   - template: The prompt template to use (overrides what is defined in the Modelfile).
     /// - Returns: A `ChatResponse` containing the generated message and additional information.
     /// - Throws: An error if the request fails or the response cannot be decoded.
     public func chat(
         model: Model.ID,
         messages: [Chat.Message],
         options: [String: Value]? = nil,
+        tools: [Chat.ToolDefinition]? = nil,
         template: String? = nil
     )
         async throws -> ChatResponse
     {
+        let toolDefinitions: [Chat.ToolDefinition] = tools ?? self.tools.map({ $0.definition })
```

Comment:
Ignoring the parameter seems a bit confusing to me.

I would either remove the tools array from the client or combine both arrays.

Reply:
The idea is the tools are global, so you don't have to keep passing them in every single time. However, I left it as a parameter so you can override which tools are sent for this specific chat request. As it has a default parameter most calls sites omit the `tools` parameter entirely, and it just works. But if you want to override that, or pass an empty array you can.